Modules/Scripted/Home/Home.py

```diff
@@ -92,6 +96,108 @@ def setup(self):
         # SPACER
         self.layout.addStretch()
 
+        #SAMPLE DATA REGISTRATION
+        json_list = [self.resourcePath('SampleDataDescription/ShapeRegressionInputData.json'),
+                     self.resourcePath('SampleDataDescription/DataImporterInputData.json'),
+                     self.resourcePath('SampleDataDescription/SPHARM-PDMTestData.json'),]
+
+        for json_file in json_list:
+            with open(json_file, 'r') as f:
+                source_data = json.load(f)
+                #Register sample datasets for each module
+                iconPath = os.path.join(os.path.dirname(__file__).replace('\\','/'), 'Resources','Icons')
+                iconPath = None
+                #Data Importer sample dataset registration
+                SampleData.SampleDataLogic.registerCustomSampleDataSource(
+                  category=source_data['category'], sampleName=source_data['sampleName'],
+                  uris=source_data['uris'],
+                  fileNames=source_data['fileNames'],
+                  nodeNames=None,
+                  thumbnailFileName=iconPath,
+                  loadFileType=None,
+                  customDownloader = self.DownloadSampleDataInFolder,
+                  checksums = source_data['checksums']
+                )
+
     def onAnchorClicked(self, url):
         moduleName = url.fragment()
         slicer.util.selectModule(moduleName)
+
+    def DownloadSampleDataInFolder(self,source):
+        print(self)
+        destFolderPath = str(qt.QFileDialog.getExistingDirectory())
+
+        if not os.path.isdir(destFolderPath):
+            print('Invalid path %s' %{destFolderPath})
+            print("Can't download the sample data")
+            return
+
+        if not hasattr(source,'checksums'):
+            checksums = [None] * len(source.uris)
+        else:
+            checksums = source.checksums
+
+        for uri, fileName, checksum  in zip(source.uris,source.fileNames,checksums):
+            self.downloadFile(uri, destFolderPath, fileName, checksum=checksum)
+
+        print('Data folder: %s' % destFolderPath)
+
+    def reportHook(self,blocksSoFar,blockSize,totalSize):
```

Comment:
Why do we need to copy the code for `reportHook`, `downloadFile` and `humanFormatSize` ?